pandas/core/arrays/base.py

```diff
@@ -787,7 +797,12 @@ def unique(self):
         uniques = unique(self.astype(object))
         return self._from_sequence(uniques, dtype=self.dtype)
 
-    def searchsorted(self, value, side="left", sorter=None):
+    def searchsorted(
+        self,
+        value: ArrayLike,
+        side: Optional[Literal["left", "right"]] = "left",
+        sorter: Optional[ArrayLike] = None,
```

Comment:
`ArrayLike` is a typevar, so the type of `value` and `sorter` must be the same otherwise I _think_ we would get mypy errors. (although we may not get errors while np.ndarray resolves to any). 

(this is another area where api testing could be useful #40202)

Reply:
I don't see how we can make them the same when we are allowing `sorter` to be `None`.  I am passing the `mypy` tests with the code as above.

Reply:
sorry, didn't explain properly. The types are saying that the types are the same if `sorter` is not None. We don;t want that.

> I am passing the `mypy` tests with the code as above.

that's because np.ndarry resolves to `Any`. the sooner #39513 is resolved (i.e #36092 gets merged, if no-one is going to suggest a practical alternative) the better. (after over 6 months on that PR, I am almost done with typing for pandas, its getting tiresome (and very time consuming) keeping #36092 up-to-date just to see more hidden errors appearing daily.